docs/architecture/adrs/002-streaming-response-aggregation.md

```diff
@@ -0,0 +1,43 @@
+# ADR 002: Streaming Response Behavior and Result Aggregation
+
+This ADR defines the orchestrator strategy of result aggregation and behavior observed by end users of the initial streaming API endpoint (i.e. streaming text generation with classification) when detections are made on streamed output of text generation models.
+
+## Motivation
+
+This orchestrator is designed to work with various detectors. Each detector will have an associated chunker indicated in an end user configuration. Each chunker could chunk different amounts of text e.g. a sentence chunker vs. a paragraph chunker. Each detector could then provide detection responses on different amounts of text. In the unary case, all of these detection responses could be aggregated together and returned in the single response to the end user. 
+
+However, in the streaming case, as frames are being streamed back, the end user will need to receive detection responses corresponding to the text in the streamed frames. Frames still have to be returned in order so that the user can parse the stream of generated text and detections and know how much of the original text generation stream has been processed i.e. what part of the stream that detections have been run on.
+
+
+## Decisions
+
+### End user experience
+
+1. Only fully processed frames will be returned to the end user. This means that for each (streamed) frame returned to the user, all detector responses for that particular frame's text up to that frame’s index (`processed_index`) will be returned with the frame.
+    1. Generated texts in output frames with detections are still expected to be contiguous, meaning the end user can still piece all of the frames together. This aligns with the way various text generation servers provide output.
+2. In order to achieve (1) for detectors that may work with different chunkers, e.g. paragraph vs. sentence, results will not be streamed back/returned until the “slowest” detector has processed. “Slowest” may not necessarily just be in terms of time but in terms of amount of text processed (i.e. largest `processed_index`).  
```

Comment:
One of the implication here is also that the "default document chunker" (or no chunker) that is within orchestrator, will also have to provide response in the datastructure of streaming 🤔